src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -39,6 +39,28 @@ impl ScalarFunc {
         Ok(Some(Cow::Owned(format!("{:b}", i).into_bytes())))
     }
 
+    pub fn left<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let s = try_opt!(self.children[0].eval_string(ctx, row));
+        let i = try_opt!(self.children[1].eval_int(ctx, row));
+        if i <= 0 {
+            return Ok(Some(Cow::Owned(b"".to_vec())));
+        }
+        {
+            let s = str::from_utf8(&s)?;
```

Comment:
I think we should call `eval_string_and_decode` instead of `str::from_utf8`. So when we support more codec in future, it can reduce lots of work.

Reply:
OK. I've fixed here. PTAL